include/RAJA/pattern/tensor/TensorIndex.hpp

```diff
@@ -29,7 +29,7 @@ namespace expt
 {
 
 
-  template<typename IDX, typename TENSOR_TYPE, camp::idx_t DIM, IDX INDEX_VALUE, strip_index_type_t<IDX> LENGTH_VALUE>
```

Comment:
Support for strongly typed indices with `TensorIndex` was missing, and that is fixed here. We don't have this in our CI testing, so I'll create an issue to add tests for that.